v2/api/network/v1alpha1api20201101/virtual_network_types_gen.go

```diff
@@ -335,112 +340,165 @@ type VirtualNetwork_Status struct {
 	//the virtual network.
 	EnableVmProtection *bool `json:"enableVmProtection,omitempty"`
 
-	//Etag: A unique read-only string that changes whenever the resource is updated.
-	Etag *string `json:"etag,omitempty"`
-
 	//ExtendedLocation: The extended location of the virtual network.
-	ExtendedLocation *ExtendedLocation_Status `json:"extendedLocation,omitempty"`
-
-	//Id: Resource ID.
-	Id *string `json:"id,omitempty"`
+	ExtendedLocation *ExtendedLocation `json:"extendedLocation,omitempty"`
 
 	//IpAllocations: Array of IpAllocation which reference this VNET.
-	IpAllocations []SubResource_Status `json:"ipAllocations,omitempty"`
+	IpAllocations []SubResource `json:"ipAllocations,omitempty"`
 
 	//Location: Resource location.
 	Location *string `json:"location,omitempty"`
 
-	//Name: Resource name.
-	Name *string `json:"name,omitempty"`
-
-	//ProvisioningState: The provisioning state of the virtual network resource.
-	ProvisioningState *ProvisioningState_Status `json:"provisioningState,omitempty"`
+	// +kubebuilder:validation:Required
+	Owner genruntime.KnownResourceReference `group:"resources.azure.com" json:"owner" kind:"ResourceGroup"`
 
-	//ResourceGuid: The resourceGuid property of the Virtual Network resource.
-	ResourceGuid *string `json:"resourceGuid,omitempty"`
+	//Reference: Resource ID.
+	Reference *genruntime.ResourceReference `armReference:"Id" json:"reference,omitempty"`
```

Comment:
Not 100% sure this is right. This looks like the ID field (which should be GET only, I think?) is being turned into a field called `Reference`. I think this just shouldn't be on the spec at all?

Reply:
yes this stuff is still messed up, it's the final keystone piece :D